docs/samples/browser-plugin-meetings/app.js

```diff
@@ -1190,34 +1343,87 @@ async function startScreenShare() {
     localMedia.screenShare.video = localShareVideoTrack;
     localMedia.screenShare.video.on('published-state-update', ({isPublished}) => {
       if (!isPublished) {
-        console.log('MeetingControls#startScreenShare() :: local share video track unpublished, stopping it');
-
-        localMedia.screenShare.video.stop();
-        localMedia.screenShare.video = undefined;
-
-        meetingStreamsLocalShare.srcObject = null;
+        console.log('MeetingControls#startScreenShare() :: local share video track unpublished');
+        // not stopping share video track as we want to use it across meetings
       }
+      isScreenshareTrackPublished = isPublished;
+      runPublishedStateTags();
     });
 
+    // this handler is for when browser's stop share button is clicked
+    localMedia.screenShare.video.underlyingTrack.onended = async () => {
+      meetingStreamsLocalShare.srcObject = null;
+      localMedia.screenShare.video = null;
+      localScreensharePublishedState.innerText = '';
+      await stopScreenShare();
+    }
+
     meetingStreamsLocalShare.srcObject = localShareVideoTrack.underlyingStream;
+    runPublishedStateTags();
+
+    console.log('MeetingControls#startScreenShare() :: Successfully got screen sharing!');
+  }
+  catch (error) {
+    console.log('MeetingControls#startScreenShare() :: Error getting screen share!');
+    console.error(error);
+  }
+}
+
+async function publishScreenShare() {
+  const meeting = getCurrentMeeting();
+  const {screenShare} = localMedia;
 
-    console.log('MeetingControls#startScreenShare() :: publishing share video track');
+  if (!meeting) {
+    console.log('MeetingControls#publishScreenShare() :: no valid meeting object!');
+
+    return Promise.reject(new Error('No valid meeting object.'));
```

Comment:
Same here, we can simply throw the error as the function is async. Please check other places too

Reply:
done